packages/travix-ui-kit/components/stepper/stepper.js

```diff
@@ -0,0 +1,135 @@
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import React, { Component } from 'react';
+
+/**
+ * General Stepper component
+ */
+class Stepper extends Component {
+  constructor(props) {
+    super(props);
+
+    this.validationFn = (stepper, triedValue) => stepper.setValue(triedValue);
+    this.state = {
+      currentValue: Math.max(Math.min(props.initValue, props.maxValue), props.minValue),
+    };
+
+    if (typeof props.validationFn === 'function') {
+      this.validationFn = props.validationFn;
+    }
+
+    this.tryIncreaseValue = this.tryIncreaseValue.bind(this);
+    this.tryDecreaseValue = this.tryDecreaseValue.bind(this);
+    this.setValue = this.setValue.bind(this);
+    this.getValue = this.getValue.bind(this);
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (newProps.initValue !== this.state.currentValue) {
+      this.setState({
+        currentValue: Math.max(Math.min(newProps.initValue, this.props.maxValue), this.props.minValue),
+      });
+    }
+  }
+
+  getValue() {
+    return this.state.currentValue;
+  }
+
+  setValue(newValue, cb) {
+    const newStateValue = Math.max(Math.min(newValue, this.props.maxValue), this.props.minValue);
```

Comment:
We should move this validation (not allowing min and max) to be checked before we try to set the value. There is no reason to try to set a value with the same value as before.